### PR TITLE
Commit: HomeWork_06_03_21_MyList

### PR DESCRIPTION
**Pull request by diledsys**

### CODE REVIEW

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package com.java_essential.diledsys.homeWork.myDictionary;
+
+public class MyDictionar<K, V> implements MyList {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:3]: public class MyDictionar<K, V> implements MyList<K, V>

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    private int size;
+    private Element head;
+    private Element tail;
+
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:7]: private Element<K, V> head;
    private Element<K, V> tail;

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            }
+            temp = temp.next;
+        }
+        return null;
+    }
+
+    @Override
+    public void put(Object key, Object value) {
+        setSize(++size);
+        Element a = new Element();
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:33]: Element<K, V> a = new Element<>();

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            if (temp.key.equals(key)) {
+                if (temp.value != null) return temp.value;
+            }
+            temp = temp.next;
+        }
+        return null;
+    }
+
+    @Override
+    public void put(Object key, Object value) {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:31]: public void put(K key, V value)

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    @Override
+    public void remove(Object key) {
+        setSize(--size);
+        if (head == null) return;
+        if (head == tail) {
+            head = null;
+            tail = null;
+        }
+        if (head.key == key) {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:53]: == on object, seriously!?
Objects.equals(head.key, key)

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            head = head.next;
+            return;
+        }
+        Element temp = head;
+        while (temp.next != null) {
+            if (temp.next.key == key) {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:59]: == on object, seriously!?
Objects.equals(head.key, key)

File: src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
+package com.java_essential.diledsys.homeWork.myList;
+
+import java.util.Arrays;
+
+public class MyArrayList<E> implements MyList {
+
+    private  Object[] array = new Object[1];
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java:7]: private  E[] array = (E[]) new Object[SIZE];
 The best option for the size of the initial array is 10.

File: src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
+package com.java_essential.diledsys.homeWork.myList;
+
+import java.util.Arrays;
+
+public class MyArrayList<E> implements MyList {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java:5]: public class MyArrayList<E implements MyList<E

File: src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean add(Object el) {
+        setSize(raise());
+        array[size] = new Element<E>((E) el);
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java:22]: For what you need this inner class? Work with array E directly.

File: src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+    private int raiseminus(int count) {
+        size = array.length - count;
+        if (size > 0) {
+            array = Arrays.copyOf(array, size);
+        }
+        return size - 1;
+    }
+
+    public class Element<E> {
```
Comment (ochedenis) [inline src/main/java/com/java_essential/diledsys/homeWork/myList/MyArrayList.java:70]: For what you need this inner class? Work with array E directly.

**Review by ochedenis — CHANGES_REQUESTED** [review]

File: src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java (hunk 5 above)
Comment (diledsys) [reply src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:53]: так key это типовое поле, разве его нельзя равенством проверить

Comment (diledsys) [reply src/main/java/com/java_essential/diledsys/homeWork/myDictionary/MyDictionar.java:53]: или желательно эквиолентность

**Pull request merged**